app/cocina_builders/related_works_cocina_builder.rb

```diff
@@ -26,16 +50,41 @@ def call
 
   def related_work_params_for(related_work:)
     {
-      type: related_work['relationship']
+      type: RELATION_TYPES.dig(related_work['relationship'], :cocina_type),
+      dataCiteRelationType: RELATION_TYPES.dig(related_work['relationship'], :datacite)
     }.compact.tap do |related_work_hash|
-      if related_work['identifier'].present?
-        related_work_hash[:identifier] =
-          [{ uri: related_work['identifier'] }]
-      end
+      related_work_hash.merge!(build_identifier(related_work['identifier']))
 
       if related_work['citation'].present?
         related_work_hash[:note] = [{ type: 'preferred citation', value: related_work['citation'] }]
       end
     end
   end
+
+  # Build identifier or access cocina fields from form identifier value if present
+  def build_identifier(identifier)
+    return {} if identifier.blank?
+
+    if PurlSupport.purl?(url: identifier)
+      { purl: identifier }
+    elsif uri?(identifier)
+      { identifier:
+            [{ uri: identifier, type: uri_type_for(identifier) }] }
+    else
+      { access: { url: [{ value: identifier }] } }
+    end
+  end
+
+  def uri?(identifier)
```

Comment:
Can this just be: `uri_type_for(identifier).present?`?

Reply:
Right, that works. Done.